packages/hub/src/lib/cache-management.ts

```diff
@@ -0,0 +1,259 @@
+			files: cachedFiles,
+			refs: refsByHash.get(dir) || [],
+			size: [...cachedFiles].reduce((sum, file) => sum + file.blob.size, 0),
+			path: revisionPath,
+			lastModifiedAt: new Date(revisionLastModified),
+		});
+
+		refsByHash.delete(dir);
+	}
+
+	// Verify that all refs refer to a valid revision
+	if (refsByHash.size > 0) {
+		throw new Error(
+			`Reference(s) refer to missing commit hashes: ${JSON.stringify(Object.fromEntries(refsByHash))} (${repoPath})`
+		);
+	}
+
+	const repoStats = await stat(repoPath);
+	const repoLastAccessed =
+		blobStats.size > 0 ? Math.max(...[...blobStats.values()].map((stat) => stat.atimeMs)) : repoStats.atimeMs;
+
+	const repoLastModified =
+		blobStats.size > 0 ? Math.max(...[...blobStats.values()].map((stat) => stat.mtimeMs)) : repoStats.mtimeMs;
+
+	// Return the constructed CachedRepoInfo object
+	return {
+		id: {
+			name: repoId,
+			type: repoType,
+		},
+		path: repoPath,
+		filesCount: blobStats.size,
+		revisions: cachedRevisions,
+		size: [...blobStats.values()].reduce((sum, stat) => sum + stat.size, 0),
+		lastAccessedAt: new Date(repoLastAccessed),
+		lastModifiedAt: new Date(repoLastModified),
+	};
+}
+
+export async function scanRefsDir(refsPath: string, refsByHash: Map<string, string[]>): Promise<void> {
+	const refFiles = await readdir(refsPath, { withFileTypes: true });
+	for (const refFile of refFiles) {
+		const refFilePath = join(refsPath, refFile.name);
+		if (refFile.isDirectory()) continue; // Skip directories
+
+		const commitHash = await readFile(refFilePath, "utf-8");
+		const refName = refFile.name;
+		if (!refsByHash.has(commitHash)) {
+			refsByHash.set(commitHash, []);
+		}
+		refsByHash.get(commitHash)?.push(refName);
+	}
+}
+
+export async function scanSnapshotDir(
+	revisionPath: string,
+	cachedFiles: CachedFileInfo[],
+	blobStats: Map<string, Stats>
+): Promise<void> {
+	const files = await readdir(revisionPath, { withFileTypes: true });
+	for (const file of files) {
+		if (file.isDirectory()) continue; // Skip directories
+
+		const filePath = join(revisionPath, file.name);
+		const blobPath = await realpath(filePath);
+		const blobStat = await getBlobStat(blobPath, blobStats);
+
+		cachedFiles.push({
+			path: filePath,
+			blob: {
+				path: blobPath,
+				size: blobStat.size,
+				lastAccessedAt: new Date(blobStat.atimeMs),
+				lastModifiedAt: new Date(blobStat.mtimeMs),
+			},
+		});
+	}
+}
+
+export async function getBlobStat(blobPath: string, blobStats: Map<string, Stats>): Promise<Stats> {
+	const blob = blobStats.get(blobPath);
+	if (!blob) {
+		const statResult = await lstat(blobPath);
+		blobStats.set(blobPath, statResult);
+		return statResult;
+	}
+	return blob;
+}
+
+export function parseRepoType(type: string): RepoType {
+	switch (type) {
+		case "models":
+		case "model":
+			return "model";
+		case "dataset":
+			return "dataset";
+		case "space":
+			return "space";
+		default:
+			throw new TypeError(`Invalid repo type: ${type}`);
+	}
```

Comment:
Did it happen to you to have the folders in the format `model--username--repo-name` / `dataset--username--repo-name` / `space--username--repo-name` ? If that's the case, it's a bug somewhere. The cache system from `huggingface_hub` is supposed to only use `models`/`datasets`/`spaces`.

```suggestion
	switch (type) {
		case "models":
			return "model";
		case "datasets":
			return "dataset";
		case "spaces":
			return "space";
		default:
			throw new TypeError(`Invalid repo type: ${type}`);
	}
```